### PR TITLE
[Hotfix][Connector][Jdbc] Fix reconnect throw close statement exception

### PR DESCRIPTION
**Pull request by hailin0**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
PTAL

**Review by ic4y — DISMISSED** [review]
LGTM

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/JdbcOutputFormat.java [hunk 1]
```diff
@@ -239,7 +239,14 @@ public synchronized void close() {
...
 
     public void updateExecutor(boolean reconnect) throws SQLException, ClassNotFoundException {
-        jdbcStatementExecutor.closeStatements();
+        try {
+            jdbcStatementExecutor.closeStatements();
+        } catch (SQLException e) {
+            if (!reconnect) {
+                throw e;
+            }
+            LOG.info("Close JDBC statement failed on reconnect.", e);
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/JdbcOutputFormat.java]: Use error level to print logs?

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/JdbcOutputFormat.java]: updated

**Review by liugddx — APPROVED** [review]

**Review by TaoZex — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**